library/core/src/ptr/mut_ptr.rs

```diff
@@ -2195,6 +2195,48 @@ impl<T> *mut [T] {
     }
 }
 
+impl<T, const N: usize> *mut [T; N] {
+    /// Returns a raw pointer to the array's buffer.
+    ///
+    /// This is equivalent to casting `self` to `*const T`, but more type-safe.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// #![feature(array_ptr_get)]
+    /// use std::ptr;
+    ///
+    /// let arr: *mut [i8; 3] = ptr::null_mut();
+    /// assert_eq!(arr.as_mut_ptr(), ptr::null_mut());
+    /// ```
+    #[inline]
+    #[unstable(feature = "array_ptr_get", issue = "119411")]
+    pub fn as_mut_ptr(self) -> *mut T {
+        self as *mut T
+    }
+
+    /// Returns a raw pointer to a unique slice containing the entire array.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(array_ptr_get)]
+    ///
+    /// let mut arr = [1, 2, 5];
+    /// let ptr: *mut [i32; 3] = &mut arr;
+    /// unsafe {
+    ///     (&mut *ptr.as_mut_slice())[..2].copy_from_slice(&[3, 4]);
+    /// }
+    /// assert_eq!(arr, [3, 4, 5]);
+    /// ```
+    #[inline]
+    #[unstable(feature = "array_ptr_get", issue = "119411")]
+    pub fn as_mut_slice(self) -> *mut [T] {
+        // SAFETY: `N` is the length of the array, so the pointer is always in-bounds.
+        unsafe { slice::from_raw_parts_mut(self.as_mut_ptr(), N) }
```

Comment:
This is incorrect and unsound, the safety comment is incorrect.
`core::slice::from_raw_parts_mut` requires the pointer to be *valid*, while this function does not, it can be called with arbitrarily invalid pointers (like a null pointer as shown in the `as_ptr` example.

You need to use https://doc.rust-lang.org/1.75.0/std/ptr/fn.slice_from_raw_parts_mut.html  instead. That function is also safe, which matches the safety of this function.

Reply:
I recommend adding a test calling this function with a null pointer to ensure that it works afterwards.